apps/coursition/src/components/fileDropper.tsx

```diff
@@ -0,0 +1,63 @@
+'use client'
+
+import { Button } from '@nmit-coursition/design-system'
+import { cn } from '@nmit-coursition/ui/utils'
+import { X as DeleteIcon } from 'lucide-react'
+import { useState } from 'react'
+import { useDropzone, type DropzoneOptions } from 'react-dropzone'
+
+interface Props extends DropzoneOptions {
+  idleMessage?: string
+  dropZoneMessage?: string
+  className?: string
+
+  // todo: improve composability with mixins
+  inputName: string
+}
+
+export const FileDropper = ({ dropZoneMessage, idleMessage, className, inputName, ...forwardedProps }: Props) => {
+  const [files, setFiles] = useState<File[]>([])
+
+  const onDropAccepted = (acceptedFiles: File[]): void => {
```

Comment:
**question:** Do we need this logic at all? Aren't the files automatically attached to input, so that we can get them via FormData?

Reply:
Well, I'll take it off.
But if we intend to use controllled inputs in the future, then we can bring it back 👍🏽